blocks/library/paragraph/index.js

```diff
@@ -183,6 +184,10 @@ registerBlockType( 'core/paragraph', {
 					onMerge={ mergeBlocks }
 					onReplace={ onReplace }
 					placeholder={ placeholder || __( 'New Paragraph' ) }
+					formattingControls={ [ 'bold', 'italic', 'strikethrough', 'link', 'red' ] }
+					formatters={ [
+						createInlineStyleFormatter( 'red', 'hammer', 'Red', { color: 'red' } ),
```

Comment:
Without looking at the function definition, it is difficult to know at a glance what these arguments represent. Could we not just pass this as an array of objects? Why do we need `createInlineStyleFormatter` at all if we can provide a plain object representation that satisfies the need?

Reply:
Or if formatters need to be keyed by a name (which we're passing as an argument), then an object of key => setting pairs.

Could we generate a name on behalf of the implementer, based on the other settings provided? Something like a [json-stable-stringify](https://github.com/substack/json-stable-stringify) on the other options to create a unique and deterministic key. Might be overkill.

Reply:
Maybe we can pass the args as an object? I foresee the formatting object getting a lot more complex like in https://github.com/WordPress/gutenberg/blob/ece64eea2e1b7bf0bddc350fb9ddb685a2718093/blocks/library/paragraph/index.js#L109 to deal with formats that involve other settings, so thought it might be better to always use a function to create a formatter?

Reply:
If the formatting object becomes complex, how does the function remedy this? If it's the case that it generates many of the values of the objects based on the arguments, then I expect we could still do similar by generating those values based on the incoming formatters prop objects when they come time to be used.

Reply:
No problems, changed this to a simple object.